File: Life/Infrastructure/GameTranslator.cs
#region Usings

using Domain;
using Cell = Infrastructure.DataObjects.Cell;

#endregion

namespace Infrastructure {
    public class GameTranslator {
        public void Persist(Game game, DataObjects.Game doGame) {
            doGame.Size = game.Size;
            doGame.Name = game.Name;
            PersistCells(game, doGame);
        }

        private void PersistCells(Game game, DataObjects.Game doGame) {
            doGame.Cells.Clear();
            for (var row = 0; row < game.Size; row++) {
                for (var column = 0; column < game.Size; column++) {
                    if (game[row, column].IsAlive) {
                        doGame.Cells.Add(new Cell {X = column, Y = row});
                    }
                }
            }
        }

        public Game Reconstitute(DataObjects.Game doGame) {
            var game = new Game(doGame.Size, doGame.Name);
            foreach (var cell in doGame.Cells) {
                game.GiveBirth(cell.Y, cell.X);
            }
            return game;
        }
    }
}

File: life-java/src/main/java/ru/agiledays/life/server/infrastructure/GameTranslator.java
package ru.agiledays.life.server.infrastructure;

import ru.agiledays.life.server.infrastructure.dataobjects.*;

public class GameTranslator {
    public void persist(final ru.agiledays.life.server.domain.Game game,
                        final Game doGame) {
        doGame.setSize(game.size());
        doGame.setName(game.name());
        persistCells(game, doGame);
    }

    private void persistCells(final ru.agiledays.life.server.domain.Game game,
                              final Game doGame) {
        doGame.getCells().clear();
        for (int row = 0; row < game.size(); row++) {
            for (int column = 0; column < game.size(); column++) {
                if (game.getCell(row, column).isAlive()) {
                    doGame.getCells().add(new Cell(row, column));
                }
            }
        }
    }

    public ru.agiledays.life.server.domain.Game reconstitute(Game doGame) {
        final ru.agiledays.life.server.domain.Game game =
                new ru.agiledays.life.server.domain.Game(doGame.getSize(), doGame.getName());
        for (Cell cell: doGame.getCells()) {
            game.giveBirth(cell.getY(), cell.getX());
        }
        return game;
    }
}
